src/vue/pages/Verification.vue

```diff
@@ -0,0 +1,206 @@
+      </p>
+    </div>
+    <p class="verification__subtitle">
+      {{ 'verification-page.account-type-lbl' | globalize }}
+    </p>
+    <div class="account-type-selector">
+      <router-link
+        :to="vueRoutes.verification.general"
+        class="account-type-selector__item"
+      >
+        <p class="account-type-selector__item-title">
+          {{ 'verification-page.account-type-general-title' | globalize }}
+        </p>
+        <p class="account-type-selector__item-description">
+          {{ 'verification-page.account-type-general-description'
+            | globalize }}
+        </p>
+        <div class="account-type-selector__selected-icon">
+          <i class="mdi mdi-check" />
+        </div>
+      </router-link>
+      <router-link
+        :to="vueRoutes.verification.corporate"
+        class="account-type-selector__item"
+      >
+        <p class="account-type-selector__item-title">
+          {{ 'verification-page.account-type-corporate-title' | globalize }}
+        </p>
+        <p class="account-type-selector__item-description">
+          {{ 'verification-page.account-type-corporate-description'
+            | globalize }}
+        </p>
+        <div class="account-type-selector__selected-icon">
+          <i class="mdi mdi-check" />
+        </div>
+      </router-link>
+    </div>
+    <div class="verification__form">
+      <router-view />
+    </div>
+  </div>
+</template>
+
+<script>
+import { vuexTypes } from '@/vuex'
+import { mapGetters } from 'vuex'
+
+import { vueRoutes } from '@/vue-router/routes'
+
+import { REQUEST_STATES_STR } from '@/js/const/request-states.const'
+
+import { globalize } from '@/vue/filters/globalize'
+
+export default {
+  name: 'verification',
+  data: _ => ({
+    vueRoutes,
+  }),
+  computed: {
+    ...mapGetters({
+      kycState: vuexTypes.kycState,
+      kycLatestData: vuexTypes.kycLatestData,
+      kycRejectReason: vuexTypes.kycRequestRejectReason,
+    }),
+    kycRequestMessageId () {
+      switch (this.kycState) {
+        case REQUEST_STATES_STR.pending:
+          return 'verification-page.pending-request-msg'
+        case REQUEST_STATES_STR.approved:
+          return 'verification-page.approved-request-msg'
+        case REQUEST_STATES_STR.rejected:
+          return 'verification-page.rejected-request-msg'
+        default:
+          return ''
+      }
+    },
+    kycRequestMessage () {
+      if (this.kycState === REQUEST_STATES_STR.rejected) {
+        return globalize(this.kycRequestMessageId, {
+          reason: this.kycRejectReason,
+        })
+      } else {
+        return globalize(this.kycRequestMessageId)
+      }
+    },
+  },
+  beforeRouteUpdate (to, from, next) {
+    if (this.kycState && to.name === vueRoutes.verification.name) {
+      if (this.kycLatestData.name) {
+        next(vueRoutes.verification.corporate)
+      } else if (this.kycLatestData.first_name) {
+        next(vueRoutes.verification.corporate)
```

Comment:
1. doesn't it make the same stuff you've written in the guard in router.js? If so, I'd use this hook instead of using the guard
2. you've probably wanted to redirect the `verification.general` here
3. what if none of the conditions pass? it will be hard to debug such case

Reply:
Resolved, updated